types/config.go

```diff
@@ -80,6 +80,9 @@ type Config struct {
 	// attach/detach.
 	ProgramPath string
 
+	// ProxmoxConfig configures various attributes about the ProxMox provider.
+	ProxmoxConfig ProxmoxConfig
```

Comment:
So this is the right path, however, since we have 15 different infrastructure targets I don't think we want to include 15 separate ones into each top level config.

I was thinking this could be implemented via an interface so each target infrastructure can have its own cloud specific config but we only have one reference to it from here.

Reply:
I'm afraid that a lot of code rework will be required in many places with the config, because everything is initially taken from c.Config, if I understand correctly. I'm not sure this is necessary, because provider selected from command line (-t arg) and if has correct error checking in providers, then user can not use wrong config. But this is my opinion, I just wanted to help get the ops working with proxmox.

Reply:
the problem is the same as the original one - this is all config for a specific target platform that can be arbitrarily large and it doesn't make sense to include for other users who won't use those platforms, i can prob take a look at this in the nxt few days as I do see your need but I just wouldn't want someone else to come in and start adding in the same thing for gcp, aws, azure, etc.

we have the notion of a Provider interface here -> https://github.com/nanovms/ops/blob/master/lepton/provider.go#L21 - I think adding the config into this so each provider can have one would be a good way to go about implementing this

Reply:
No problem. It's up to you, I can wait if you rewrite it the way you want it yourself. You will make what architecture you need yourself for processing configs. When you implement, just reject the pull request and take what is required from my repository, that I wrote for proxmox and even that you can change to your liking.

Reply:
so as an example would this work for you? https://github.com/nanovms/ops/compare/master...targetConfig - that way that config is re-usable by any target platform and can be arb. large, in the past when we've had a number of target specific vars (such as Azure) we've utilized env vars which would also be an option here

Reply:
Yes, good solution. Big Thanks.